### PR TITLE
Used parse_time function to compare Time

### LINKED ISSUE #3839: Error parsing time in VSO search filtering
### Description

VSO fails to filter results having Time as Attribute. The bug lies in the imoplementation for filtering time specifically in the line (in `vso/attrs.py`) - 
`attr.min <= _attrs.Time.strptime(it.time.end, _TIMEFORMAT)`
The comparison is occuring between a string and a Time object. Moreover, there is no attribute named `strptime` in Time attribute. 

### Expected behavior

The Time attribute would have been successfully filtered with the new query.

### Actual behavior

See gist link below.

### Steps to Reproduce

Gist link - https://gist.github.com/aribalam/4aef458ebaeeb7cb23d45720f61ced2f

### PR DESCRIPTION
**Pull request by aribalam**
### Description

Removed the old code for filtering Time attribute in `vso/attrs.py`. Converted the time string (`it.time.start` and `it.time.end`) to Time objects using `parse_time` helper function.

Fixes #3839

### CODE REVIEW

File: sunpy/net/vso/attrs.py [hunk 1]
```diff
@@ -353,11 +355,11 @@ def _(attr, results):
         if
         it.time.end is not None
         and
-        attr.min <= _attrs.Time.strptime(it.time.end, _TIMEFORMAT)
+        attr.min <= parse_time(it.time.end)
```
Comment (vn-ki) [inline sunpy/net/vso/attrs.py]: What was the use of `TIMEFORMAT`?

Comment (vn-ki) [reply sunpy/net/vso/attrs.py]: If it's not used anywhere else, remove it.

Comment (aribalam) [reply sunpy/net/vso/attrs.py]: `_TIMEFORMAT` is being used in the converter in the same file.

Comment (vn-ki) [inline sunpy/net/vso/attrs.py]: Given we already know the format, I think using `Time.strptime` might be actually better here.

Comment (ayshih) [reply sunpy/net/vso/attrs.py]: I just pointed out in https://github.com/sunpy/sunpy/issues/3839#issuecomment-594110352 that the code was originally `astropy.time.Time`, but got accidentally changed to `_attrs.Time` in a bad edit, which was why it broke.

Comment (Cadair) [reply sunpy/net/vso/attrs.py]: yeah I agree, we don't need to use `parse_time` we just need to undo the change as @ayshih pointed out in the issue.

Comment (aribalam) [reply sunpy/net/vso/attrs.py]: @vn-ki Check my comment in https://github.com/sunpy/sunpy/issues/3839.

File: changelog/3840.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Used the `parse_time` function to compare time in VSO post-search filtering code
```
Comment (Cadair) [inline changelog/3840.bugfix.rst]: This changelog is now inaccurate.

Comment (nabobalis) [reply changelog/3840.bugfix.rst]: Fixed.

**Comment by nabobalis** [comment]
Thanks @aribalam

**Pull request merged**